test/derivative/ClassicDeriv.cpp

```diff
@@ -31,17 +30,19 @@ ClassicDeriv::ClassicDeriv() {
       this->prodId[i_rxn][i_prod] = rand() % this->numSpec;
   }
 }
-void ClassicDeriv::Solve(const double *const state, double *const deriv) {
-  for (int i_spec = 0; i_spec < NUM_SPEC; ++i_spec)
-    deriv[i_spec] = 0.0;
-  for (int i_rxn = 0; i_rxn < this->numRxns; ++i_rxn) {
-    double rate = this->rateConst[i_rxn];
-    for (int i_react = 0; i_react < this->numReact[i_rxn]; ++i_react)
-      rate *= state[this->reactId[i_rxn][i_react]];
-    for (int i_react = 0; i_react < this->numReact[i_rxn]; ++i_react)
-      deriv[this->reactId[i_rxn][i_react]] -= rate;
-    for (int i_prod = 0; i_prod < this->numProd[i_rxn]; ++i_prod)
-      deriv[this->prodId[i_rxn][i_prod]] += rate;
+void ClassicDeriv::Solve(const double *const rateConst, const double *const state, double *const deriv) {
```

Comment:
It seems you add a new argument `rateConst` here. Could you briefly explain why you need it and how it is related to the multi-gridcell functionality?

Reply:
This is closer to how the real function in a chemistry solver would be set up. The rate constants depend on environmental conditions, so they vary cell-by-cell and in time. Before, when we were just mimicking solving one cell and one time step, I just assumed fixed rate constants. Now, with multiple cells, I thought it would make more sense to pass them in to the function.

Reply:
@mattldawson Thanks for your clarification. That makes sense.